commit 989112806d67c4bf88c46ed5d154bf2baa3a779b
Author: Cameron Watters <cameron.watters@joyent.us>
Date:   2019-03-21T10:19:24-07:00 (7 months ago)
    
    MANTA-3440 Add support for Content-Encoding for objects
    
    (Changes initially done by @IanWyszynski)

diff --git a/lib/moray.js b/lib/moray.js
index 391c079..2408cdb 100644
--- a/lib/moray.js
+++ b/lib/moray.js
@@ -375,6 +375,8 @@ function createMetadata(options) {
 
     switch (options.type) {
     case 'object':
+        assert.optionalString(options.contentEncoding,
+                'options.contentEncoding');
         assert.number(options.contentLength, 'options.contentLength');
         assert.string(options.contentMD5, 'options.contentMD5');
         assert.string(options.contentType, 'options.contentType');
@@ -388,6 +390,7 @@ function createMetadata(options) {
             });
         }
 
+        md.contentEncoding = options.contentEncoding;
         md.contentLength = options.contentLength;
         md.contentMD5 = options.contentMD5;
         md.contentType = options.contentType;
